### PR TITLE
docs: improve "Hosting with Apache"

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- improve document structure
- fix the project folder location for the configuration

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
The current doc says to set the document root to `/opt/lamp/apache2/htdocs/myproject/public`,
and `/opt/lamp/apache2/htdocs` is the document root of the main server.

This is not good.

```
<Directory "/opt/lamp/apache2/htdocs">
    Options Indexes FollowSymLinks
    AllowOverride All
    Require all granted
</Directory>
```
```
<VirtualHost *:80>
    DocumentRoot "/opt/lamp/apache2/htdocs/myproject/public"
    ServerName myproject.local
    ErrorLog "logs/myproject-error_log"
    CustomLog "logs/myproject-access_log" common
</VirtualHost>
```
https://codeigniter4.github.io/CodeIgniter4/installation/running.html#virtual-hosting

**Comment by kenjis** [comment]
Changed the location of the project folder: https://github.com/codeigniter4/CodeIgniter4/pull/7679/commits/3bab114bdbf96587112c0759d82447b61942a0fd

File: user_guide_src/source/installation/running.rst [hunk 1]
```diff
@@ -83,156 +86,200 @@ The local development server can be customized with three command line options:
...
-                </LimitExcept>
+            AllowOverride All
+            Options MultiViews Indexes FollowSymLinks
+            <Limit GET POST OPTIONS>
+                # Apache <= 2.2:
+                # Order allow,deny
+                # Allow from all
+
+                # Apache >= 2.4:
+                Require all granted
```
Comment (MGatner) [inline user_guide_src/source/installation/running.rst:230]: Ah I see, it looks like this is a new approach. I trust you on this.

**Review by MGatner — APPROVED** [review]

File: user_guide_src/source/installation/running.rst (hunk 1 above)
Comment (kenjis) [reply user_guide_src/source/installation/running.rst:230]: What do you mean? The change here is simply a change in indentation.

File: user_guide_src/source/installation/running.rst [hunk 2]
```diff
@@ -83,156 +86,200 @@ The local development server can be customized with three command line options:
...
-        ErrorLog "logs/myproject-error_log"
-        CustomLog "logs/myproject-access_log" common
+        DocumentRoot "/opt/lamp/apache2/myproject/public"
+        ServerName   myproject.local
+        ErrorLog     "logs/myproject-error_log"
+        CustomLog    "logs/myproject-access_log" common
+
+        <Directory "/opt/lamp/apache2/myproject/public">
+            AllowOverride All
+            Require all granted
```
[the remark this answers is not in the document]
Comment (kenjis) [reply user_guide_src/source/installation/running.rst:172]: Yes, it was documented:

> If your project folder is not a subfolder of the Apache document root, then your
``<VirtualHost>`` element may need a nested ``<Directory>`` element to grant the web server access to the files.

`/opt/lamp/apache2/htdocs/myproject/public` is in the main document root `/opt/lamp/apache2/htdocs`.
It is not good. So I moved it.

File: user_guide_src/source/installation/running.rst [hunk 3]
```diff
@@ -83,156 +86,200 @@ The local development server can be customized with three command line options:
...
+
+.. code-block:: apache
 
     LoadModule vhost_alias_module modules/mod_vhost_alias.so
 
+Adding Host Alias
+-----------------
+
 Add a host alias in your "hosts" file, typically **/etc/hosts** on unix-type platforms,
 or **c:/Windows/System32/drivers/etc/hosts** on Windows.
```
[the remark this answers is not in the document]
Comment (kenjis) [reply user_guide_src/source/installation/running.rst]: Good catch! Fixed.

**Pull request merged**